### PR TITLE
unix,win: add uv_os_getppid()

### PR DESCRIPTION
**Pull request by cjihrig**
Requested in https://github.com/nodejs/node/issues/14957. The only real work is on Windows, and libuv already had the functionality internally, so I thought I'd take a shot at exposing it.

CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/542/

### CODE REVIEW

File: src/win/pipe.c [hunk 1]
```diff
@@ -1969,7 +1969,7 @@ int uv_pipe_open(uv_pipe_t* pipe, uv_file file) {
 
   if (pipe->ipc) {
     assert(!(pipe->flags & UV_HANDLE_NON_OVERLAPPED_PIPE));
-    pipe->pipe.conn.ipc_pid = uv_parent_pid();
+    pipe->pipe.conn.ipc_pid = (int) uv_os_getppid();
```
Comment (bnoordhuis) [inline src/win/pipe.c]: No cast necessary if `uv_pid_t` is a typedef for `in`, right?

File: test/test-platform-output.c [hunk 2]
```diff
@@ -144,5 +145,9 @@ TEST_IMPL(platform_output) {
   printf("  shell: %s\n", pwd.shell);
   printf("  home directory: %s\n", pwd.homedir);
 
+  ppid = uv_os_getppid();
+  ASSERT(ppid != 0);
```
Comment (bnoordhuis) [inline test/test-platform-output.c]: `ppid > 0`?

File: src/win/util.c [hunk 3]
```diff
@@ -331,7 +331,7 @@ uint64_t uv_get_total_memory(void) {
 }
 
 
-int uv_parent_pid(void) {
+uv_pid_t uv_os_getppid(void) {
   int parent_pid = -1;
   HANDLE handle;
   PROCESSENTRY32 pe;
```
[the remark this answers is not in the document]
Comment (cjihrig) [reply src/win/util.c]: TBH, I'm not sure under what conditions this can fail.

**Comment by cjihrig** [comment]
Addressed nits. New CI with a couple unrelated failures: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/544/

File: src/win/util.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (santigimeno) [reply src/win/util.c]: Would be good to know so documentation can reflect it. Maybe @bzoz or @refack know better

**Pull request merged**